src/main/java/globalquake/ui/stationselect/SearchBar.java

```diff
@@ -0,0 +1,368 @@
+                        timer.stop();
+                    }
+                }
+                else{
+                    if(location<getWidth()){
+                        location+=speed;
+                        repaint();
+                    }
+                    else{
+                        timer.stop();
+                    }
+                }
+            }
+        });
+
+    }
+
+    @Override
+    protected void paintComponent(java.awt.Graphics g) {
+        int width = getWidth();
+        int height = getHeight();
+        setPreferredSize(new Dimension(200, 200));
+        Graphics2D g2 = (Graphics2D) g;
+        g2.setRenderingHint(java.awt.RenderingHints.KEY_ANTIALIASING, java.awt.RenderingHints.VALUE_ANTIALIAS_ON);
+        g2.setRenderingHint(RenderingHints.KEY_INTERPOLATION, RenderingHints.VALUE_INTERPOLATION_BILINEAR);
+        g2.setColor(backgroundColor);
+        setPreferredSize(new Dimension(width, height));
+        g2.fillRoundRect(0, 0, width, height, height, height);
+        super.paintComponent(g);
+        //Create Button
+        int marginButton = 5;
+        int buttonSize = height - marginButton * 2;
+        GradientPaint gp = new GradientPaint(0, 0, new Color(255,255,255,0), width, 0, animationColor);
+        g2.setPaint(gp);
+        g2.fillOval((width - height)+3, marginButton, buttonSize, buttonSize);
+
+        //Attempting to change the color of the country borders
+        if (countryFound && foundCoordinates != null) {
+            g2.setColor(Color.GREEN);
+            g2.setStroke(new BasicStroke(2)); 
+            ArrayList<Polygon> paths = new ArrayList<>();
+
+            if(foundCoordinates instanceof org.geojson.Polygon){
+                Polygon pol = (org.geojson.Polygon) foundCoordinates;
+                paths.add(pol);
+            }
+            else if(foundCoordinates instanceof org.geojson.MultiPolygon){
+                
+                MultiPolygon mp = (org.geojson.MultiPolygon) foundCoordinates;
+                List<List<List<org.geojson.LngLatAlt>>> polygons = mp.getCoordinates();
+                for (List<List<org.geojson.LngLatAlt>> polygon : polygons) {
+                    org.geojson.Polygon pol = new org.geojson.Polygon(polygon.get(0));
+                    paths.add(pol);
+                }
+            }
+            g2.setColor(Color.GREEN);
+            for (Polygon polygon : paths) {
+                List<org.geojson.LngLatAlt> coordinates = polygon.getExteriorRing();
+                int[] xPoints = new int[coordinates.size()];
+                int[] yPoints = new int[coordinates.size()];
+                for (int i = 0; i < coordinates.size(); i++) {
+                    org.geojson.LngLatAlt point = coordinates.get(i);
+                    xPoints[i] = (int) point.getLongitude();
+                    yPoints[i] = (int) point.getLatitude();
+                }
+                g2.fillPolygon(xPoints, yPoints, coordinates.size());
+            }
+        }
```

Comment:
Here I am attempting to redraw the nation that has been searched with a different color in order to highlight that nation. The goal is to make the searched country more easily identifiable. While this code runs without errors, it does not change the nation's color and I need help debugging. I believe the reason behind the color not changing is that some classes (I believe `GlobeRenderer.java`) needs to be refreshed or revalidated in some manner.

Reply:
I see what you are trying to achieve, but this filling method will definitely not work, because the raw geojson polygon coordinates are saved as latitude/longitude and in order to calculate the actual screen coordinates, the polygon has to go trough 3D projection matrix etc...

Reply:
I would suggest working improving the UI/other things now before I modify the rendering engine a bit to allow setting properties to the individual render entities. My idea is that each render entity would have a map of properties, and you can for example create a property called `isHighlighted` and use it while rendering to set maybe different colors/thicker line.

Also I want to work soon on the events zooming / cinema mode, so your country zooming might use the same API.